Add Episode 1 exercises (#23)
czi-catalystproject/hub-champion-training: episodes/introduction.md

:::{grid-item}
Objectives

> - identify the benefits of a hub for interactive cloud computing provides

Reviewer: ```suggestion
- identify the benefits provided by a hub for interactive cloud computing
```

[See my comment on #22](https://github.com/czi-catalystproject/hub-champion-training/issues/22#issuecomment-1931564192)

## Introduction

Placeholder text.

:::{image} ../media/episodes/introduction/hub-overview.png
:width: 100%

> :alt: Diagram showing an overview of a 2i2c Hub, includiing authentication, online content, custom environments, the community hub, cloud infrastructure, support and services.

Reviewer: ❤️ alt-text

Reviewer: ```suggestion
:alt: Diagram showing an overview of a 2i2c Hub, including authentication, online content, custom environments, the community hub, cloud infrastructure, support and services.
```

:::

## Exercises

::::{dropdown} Which of the following statements are True/False?
:open:

> a. The infrastructure for running a 2i2c-managed hub is closed source and not replicable across different cloud providers.  
> b. Access to the hub can be controlled with authentication and authorisation.  
> c. Software environments are inflexible and irreproducible.  
> d. 2i2c provides operations support for your hub.

Reviewer: ```suggestion
a. 2i2c-managed hubs are only available via a single cloud provider.  
b. Access to the hub can be controlled with authentication and authorisation.  
c. Software environments are inflexible and irreproducible.  
d. 2i2c provides operations support for your hub.
```

Reviewer: ```suggestion
a. The infrastructure for running a 2i2c-managed hub is closed source and not replicable across different cloud providers.  
b. A hub can be accessed and used by anyone who knows its URL.  
c. Software environments are inflexible and irreproducible.  
d. 2i2c provides operations support for your hub.
```

Reviewer: A couple of suggestions to make these answer feel slightly more difficult to guess. Feel free to discard, and note that my changes flip the truth of both answers.

Reviewer: Not sure what will happen if you apply both of the above suggestions: maybe I should have created a single suggestion containing both changes...?

Author: I like both of your suggestions! The two suggestions will override each other, so I'll incorporate both changes into a single suggestion below and add that to a commit.

Author: ```suggestion
a. 2i2c-managed hubs are only available via a single cloud provider.  
b. A hub can be accessed and used by anyone who knows its URL.  
c. Software environments are inflexible and irreproducible.  
d. 2i2c provides operations support for your hub.
```

:::{dropdown} Solution
a. False – Hubs are built entirely with open source and community-driven tooling. The Right to Replicate gives communities the right to replicate their infrastructure in its entirety elsewhere.  
b. True – Hub Champions can control who can access the hub through GitHub and other authentication methods.  
c. False – Software environments are managed by Hub Champions and automatically deployed with containerisation, allowing for scalability across the cloud and reproducibile user environments.  
d. True – 2i2c engineers maintain service availability, uptime and operational upgrades. Hub Champions are responsible for hub configuration and management to support their user community.

:::

::::

:::{dropdown} What are the needs of your user community?
:open:

> In pairs, discuss and share the following points:

Reviewer: Share how? Will they take notes in a shared doc? Summarise their partner's answers in a report-out afterwards? I recommend to be specific about the format/structure of the exercise. If nothing else, it will affect the time required for the activity.

Author: Resolved in a97784112952652c2a394deec9be71139592e2dd

> b. True – Hub Champions can control who can access the hub through GitHub and other authentication methods.  
> c. False – Software environments are managed by Hub Champions and automatically deployed with containerisation, allowing for scalability across the cloud and reproducibile user environments.  
> d. True – 2i2c engineers maintain service availability, uptime and operational upgrades. Hub Champions are responsible for hub configuration and management to support their user community.
> 
> :::
> 
> ::::
> 
> :::{dropdown} What are the needs of your user community?
> :open:
> 
> In pairs, discuss and share the following points:

Author: ```suggestion
b. False – Hub Champions can control who can access the hub through GitHub or other authentication methods.  
c. False – Software environments are managed by Hub Champions and automatically deployed with containerisation, allowing for scalability across the cloud and reproducible user environments.  
d. True – 2i2c engineers maintain service availability, uptime and operational upgrades. Hub Champions are responsible for hub configuration and management to support their user community.

:::

::::
```

Updating the answer to statement b. following the suggestion above.

- What domain of expertise does your community have?
- How large is your community? Do they work alone or do they need to collaborate together?
- Does your community work with large datasets?

> - Does your community need access to intense computational power?

Reviewer: I anticipate some Hub Champions will report heterogeneity within their communities for these answers. Will that make a difference to the discussion? If so, should we try to capture it with an additional prompt here, e.g. _"Where applicable, are the needs for collaboration, large datasets, and/or computational power consistent across your whole community?_ If not, then no need to change.

Author: ```suggestion
- How large is your community? Do they work alone or do they need to collaborate together?
- Does your community work with large datasets?
- Does your community need access to intense computational power?
- Where applicable, are the needs for collaboration, large datasets, and/or computational power consistent across your whole community?
```

I like this discussion prompt! I definitely anticipate that this will be the case and it would be good to capture this information so that we can better serve these needs.

- What are the main software applications your community uses to conduct their work? 
- How familiar is your user community with version control using git/GitHub?

> 
> :::

Author: ```suggestion

Prepare to summarise and share with the rest of the workshop.
:::
```
I've gone with a call to action to summarise and report afterwards in response to [this comment.](https://github.com/czi-catalystproject/hub-champion-training/pull/23/files#r1481169559)

> ::::{dropdown} Which of the following are your responsibilities as Hub Champion?
> :open:
> 
> a. Providing community guidance on best research software practices for users of your hub.  
> b. Troubleshooting and supporting common user issues.  
> c. Communicating infrastructure level requests and incidents to 2i2c.  
> d. Overseeing user access policy to the hub.
> 
> :::{dropdown} Solution
> 
> All of the above are the responsibilities of a Hub Champion.

Reviewer: Is _"responsibilities"_ the best choice of word here? I worry that it might feel like formalising the role too much, and suggest:

```suggestion
::::{dropdown} In which of the following ways should a Hub Champion support their users?
:open:

a. Providing community guidance on best research software practices for users of your hub.  
b. Troubleshooting and supporting common user issues.  
c. Communicating infrastructure level requests and incidents to 2i2c.  
d. Overseeing user access policy to the hub.

:::{dropdown} Solution

All of the above are things a Hub Champion should do to enable their community to make best use of a hub.
```

Reviewer: Happy to discuss this one on our next coworking call if needed.

Author: Thanks for the tone check on this! I agree with your suggestions :)
